opm/grid/cpgrid/CpGridData.hpp

```diff
@@ -852,7 +867,11 @@ class CpGridData
     /** Level-grid or Leaf-grid cell to parent cell and refined-cell-in-parent-cell index (number between zero and total amount
         of children per parent (cells_per_dim[0]_*cells_per_dim_[1]*cells_per_dim_[2])). Entry is -1 when cell has no father. */
     std::vector<int> cell_to_idxInParentCell_;
-    
+    // Only relevant for refined level grids.
+    /** @brief Map from global_cell_lgr to leaf index set. Only relevant for refined level grids. */
+    std::unordered_map<int,int> globalCellLevel_to_leafIdx_;
```

Comment:
This should not be a member of CpGridData. If people need then this is only for a short time. Then this should be setup outside of the grid..

Take a look at:
- grid/utility/cartesianToCompressed.cpp
- grid/utility/cartesianToCompressed.hpp
- grid/utility/compressedToCartesian.cpp
- grid/utility/compressedToCartesian.hpp

Reply:
It might be time to take another look to OPM/opm-grid#687 too. Thanks for your feedback!